client/src/Component/SMSModal.js

```diff
@@ -1,23 +1,63 @@
-/* eslint-disable jsx-a11y/label-has-for */
-import { useState } from "react";
+
+import { useState, useEffect } from "react";
 import Modal from "react-bootstrap/Modal";
 import Button from "react-bootstrap/Button";
 
+
+let defaultMessage = "Hi, this is BikesforRefugees. Bikes are now available for you to pick up. Please click LINK to choose a pick up date. For any questions, please contact 07900000000.";
+
 function SMSModal(props) {
     const [show, setShow] = useState(false);
+
+    const [message, setMessage] = useState(() => {
+      const localMessage = localStorage.getItem("message");
+      if (localMessage) {
+        return localMessage;
+      }
+      return defaultMessage;
+    });
+
+
+    useEffect(() => {
+      localStorage.setItem("message", message);
+    }, [message]);
+
+    const handleChange = (e) => {
+      setMessage(e.target.value);
+    };
+
     const hideModal = () => setShow(false);
     const showModal = () => setShow(true);
 
-
     function handleSubmit () {
-      console.log(props.checkedCheckboxes);
-      hideModal();
-      }
+
+      const newMessageRequest = {
+          ids : props.selectedClients,
+          message,
+        };
+
+      fetch ("/api/send-messages", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(newMessageRequest),
+        })
+        .then((response) => {
+          if (response.status >= 200 && response.status < 300) {
```

Comment:
With fetch requests, we typically don't handle the success and error paths in one `then`. Instead, we usually handle the success path in a `then` and the error path in a `catch`, and we usually have an initial `then` check the response which throws an error if it's not okay. See the first example on the [fetch MDN page](https://developer.mozilla.org/en-US/docs/Web/API/fetch#examples) for an example of this structure (or see the fetch in `Home.js`). Let's update the code like this. Additionally, if we're hiding the modal in both the success and error paths, we can put it in a `finally` block.